apps/studio/components/interfaces/Database/Schemas/Schemas.utils.ts

```diff
@@ -12,14 +12,14 @@ const NODE_SEP = 25
 const RANK_SEP = 50
 
 export async function getGraphDataFromTables(
-  ref: string,
-  schema: PostgresSchema,
-  tables: PostgresTable[]
+  ref?: string,
+  schema?: PostgresSchema,
+  tables?: PostgresTable[]
 ): Promise<{
   nodes: Node<TableNodeData>[]
   edges: Edge[]
 }> {
-  if (!tables.length) {
+  if (!tables?.length) {
```

Comment:
this might cause a client error if tables is undefined (`undefined.length`)
initialize table as an empty array?